system-tests/ft-suite/src/test/kotlin/com/hotels/styx/routing/LoadBalancingGroupSpec.kt

```diff
@@ -0,0 +1,491 @@
+                        .let {
+                            it.value() shouldMatch "app-A-0[12]".toRegex()
+                            it.path().get() shouldBe "/"
+                            it.httpOnly().shouldBeTrue()
+                            it.maxAge().isPresent.shouldBeTrue()
+                            it.maxAge().get() shouldBe (100L)
+                        }
+            }
+
+            scenario("Responds without sticky session cookie when sticky session is not enabled") {
+                client.send(get("/normal/").header(HOST, styxServer().proxyHttpHostHeader()).build())
+                        .wait()
+                        .cookie("styx_origin_app") shouldBe Optional.empty()
+            }
+
+            scenario("Routes to origins indicated by sticky session cookie") {
+                (1..10).forEach {
+                    client.send(get("/sticky/")
+                            .header(HOST, styxServer().proxyHttpHostHeader())
+                            .cookies(requestCookie("styx_origin_stickyApp", "app-A-02"))
+                            .build())
+                            .wait()
+                            .let {
+                                it.status() shouldBe OK
+                                it.bodyAs(UTF_8) shouldBe ("mock-server-02")
+                            }
+                }
+            }
+
+            scenario("Routes to origins indicated by sticky session cookie when other cookies are provided") {
+                (1..10).forEach {
+                    client.send(get("/sticky/")
+                            .header(HOST, styxServer().proxyHttpHostHeader())
+                            .cookies(
+                                    requestCookie("other_cookie1", "foo"),
+                                    requestCookie("other_cookie2", "bar"),
+                                    requestCookie("styx_origin_stickyApp", "app-A-02"))
+                            .build())
+                            .wait()
+                            .let {
+                                it.status() shouldBe OK
+                                it.bodyAs(UTF_8) shouldBe ("mock-server-02")
+                            }
+                }
+
+            }
+
+            scenario("Routes to new origin when the origin indicated by sticky session cookie does not exist") {
+                client.send(get("/sticky/")
+                        .header(HOST, styxServer().proxyHttpHostHeader())
+                        .cookies(
+                                requestCookie("styx_origin_stickyApp", "NA"))
+                        .build())
+                        .wait()
+                        .let {
+                            it.status() shouldBe OK
+                            it.bodyAs(UTF_8) shouldMatch ("mock-server-0.")
+                        }
+            }
+
+//            scenario("!Routes to new origin when the origin indicated by sticky session cookie is no longer available") {
```

Comment:
Remove commented code and replace with issues if necessary.

Reply:
Raised #434.

Reply:
I'll leave this in as a comment:

```
  scenario("!Routes to new origin when the origin indicated by sticky session cookie is no longer available") {
     // See bug: https://github.com/HotelsDotCom/styx/issues/434
  }
```